src/Tools/dotnet-dump/ExtensionCommands/ExtensionCommandBase.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Repl;
+
+namespace Microsoft.Diagnostic.Tools.Dump.ExtensionCommands
+{
+    public abstract class ExtensionCommandBase : CommandBase
```

Comment:
Do we really need ExtensionCommandBase? Does it add that much value?  I think having the ClrMDHelper property in each of your new commands kind of makes it clearer that it is a service.  You can add the Write() function to CommandBase and having a InvokeHelp() function in each new command that has detailed help isn't much different that overriding GetDetailedHelp().

Reply:
It makes sense to me because they will be shared by all 10+ commands :^)
In addition, having the help as a simple strings is kind of explicit and hides the implementation details that might change in the future (I don't know what will be the extensibility mechanism so I'm trying to avoid any reference to underlying today's implementation)

Reply:
It's not that big of a deal and you make a good point about abstracting the current underlying extensibility for help, etc. 

But I still think Write() should be moved to CommandBase.

Reply:
You're right @mikem8361: I'm moving Write() to CommandBase